bt_ddos_shield/miner_shield.py

```diff
@@ -45,30 +50,221 @@ def __init__(self, validators_manager: AbstractValidatorsManager, address_manage
+        self.address_manager = address_manager
+        self.manifest_manager = manifest_manager
+        self.blockchain_manager = blockchain_manager
+        self.state_manager = state_manager
+        self.event_processor = event_processor
+        self.options = options
+
+        self.worker_thread = None
+        self.task_queue = Queue()
+        self.run = False
+        self.finishing = False
 
     def enable(self):
         """
-        Enable shield. It asynchronously starts the shield, which consists of such steps:
+        Enable shield. It starts worker thread, which will do such steps if run for the first time:
         1. Fetch validators keys.
         2. Creates addresses for all validators.
         3. Save manifest file.
         4. Publish link to manifest file to blockchain.
         5. Eventually close public access to original IP after some time.
 
-        It puts events to event_manager after each step. Current state is managed by state_manager. If shielding
-        process had been interrupted it is continued from the last step.
+        It puts events to event_manager after each finished operation. Current state is managed by state_manager.
+        If any error occurs it is retried forever until shield is disabled.
 
-        When shield is running, changing validators set triggers shield reconfiguration.
+        When shield is running, user can schedule tasks to be processed by worker.
         """
-        pass
+        if self.worker_thread is not None:
+            raise Exception("Shield is already enabled")
+
+        self.finishing = False
+        self.run = True
+        self._add_task(MinerShieldInitializeTask())
+        self.worker_thread = threading.Thread(target=self._worker_function)
+        self.worker_thread.start()
+
+    def disable(self):
+        """
+        Disable shield. It stops worker thread after finishing current task. Function blocks until worker is stopped.
+        """
+        self._add_task(MinerShieldDisableTask())
+        self.finishing = True
+        self.worker_thread.join()
+        self.worker_thread = None
+        self.task_queue = Queue() # clear task queue
 
     def ban_validator(self, validator_hotkey: Hotkey):
         """
-        Ban a validator by its hotkey. Function blocks execution until manifest file is updated and info about file
-        is published to Bittensor.
+        Ban a validator by its hotkey. Task will be executed by worker. It will update manifest file and publish info
+        about new file version to blockchain.
 
         Args:
             validator_hotkey: The hotkey of the validator.
         """
+        self._add_task(MinerShieldBanValidatorTask(validator_hotkey))
+        pass
+
+    def _add_task(self, task):
+        """
+        Add task to task queue. It will be handled by _worker_function.
+        """
+        if not isinstance(task, MinerShieldTask):
+            raise Exception("Task is not instance of MinerShieldTask")
```

Comment:
```suggestion
    def _add_task(self, task: MinerShieldTask):
        """
        Add task to task queue. It will be handled by _worker_function.
        """
```
Traditionally we don't implement type enforcement in the first lines of every method and function. Rather than running it in runtime, we have external tools for that which perform static analysis (whenever possible).